### ISSUE TITLE
"helm status <release> -o yaml" should have a structured resource record

### ISSUE THREAD

**Issue by ensonic** [helm/helm#5335]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.12.1", GitCommit:"02a47c7249b1fc6d8fd3b94e6b4babf9d818144e", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.1", GitCommit:"02a47c7249b1fc6d8fd3b94e6b4babf9d818144e", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:08:12Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.5-gke.5", GitCommit:"9aba9c1237d9d2347bef28652b93b1cba3aca6d8", GitTreeState:"clean", BuildDate:"2018-12-11T02:36:50Z", GoVersion:"go1.10.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
GKE

The output of `helm status cert-manager -o yaml` is
```
name: cert-manager
namespace: default
info:
  Description: Upgrade complete
  first_deployed:
    nanos: 835062970
    seconds: 1549356130
  last_deployed:
    nanos: 41466415
    seconds: 1550578181
  status:
    code: 1
    notes: |
      cert-manager has been deployed successfully!
      ...
    resources: |+
      ==> v1/ServiceAccount
      NAME           SECRETS   AGE
      cert-manager   1         14d

      ==> v1beta1/Deployment
      NAME           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
      cert-manager   1         1         1            1           14d

      ==> v1/Pod(related)
      NAME                            READY   STATUS    RESTARTS   AGE
      cert-manager-7d4bfc44ff-v4gck   1/1     Running   2          14d
```
This requires apps to manually parse the `info.resources` to see if everything is up. I look for a way to have helm tell me if the resources that belong to a release are up-and-running and that in a machine readable way. I don't think `info.status.code == 1` is sufficient or is it?

**Comment by hickeyma** [comment]
The [documentation](https://github.com/helm/helm/blob/master/docs/helm/helm_status.md) refers to the state of a release having a constant field as follows: `state of the release (can be: UNKNOWN, DEPLOYED, DELETED, SUPERSEDED, FAILED or DELETING)` .

The notes provide details alright but I think there is a need to provide a state field (e.g. `type`) which maps the code to the constant list shown above.

@bacongobbler What do you think? If it is a runner, I am willing to add it.

**Comment by bacongobbler** [comment]
Unfortunately this is a side-effect of marshalling the gRPC response straight to YAML. The output is completely intentional, as that's the exact response Helm receives from Tiller from a GetStatusRequest. For anything other than the human-readable output format (the default output of `helm status`), it's up to the end-user to parse that information into something that makes sense for them.

According to the code, it looks like status code 1 would mean the deployment was successful:

https://github.com/helm/helm/blob/8be42bae885a04b4acc242cf420911145b32ee1c/pkg/proto/hapi/release/status.pb.go#L16-L49

Given that this is a limitation of marshaling the gRPC response straight to YAML/JSON, I don't think this is something we can fix in Helm 2 without making breaking changes to the gRPC response. However, this has already been fixed in Helm 3 because we've already removed the gRPC API and instead are [generating the status strings](https://github.com/helm/helm/blob/ba622c69419aed463335caeeb63db9d8b0229a64/pkg/hapi/release/status.go#L19) internally, so `helm status -o yaml` should display the right output moving forward.
